code/modules/power/supermatter/supermatter.dm

```diff
@@ -200,102 +197,204 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
-	pressure_bonus_derived_constant = 1 / pressure_bonus_max_multiplier - pressure_bonus_derived_steepness
-	powerloss_linear_threshold = sqrt(POWERLOSS_LINEAR_RATE / 3 * POWERLOSS_CUBIC_DIVISOR ** 3)
-	powerloss_linear_offset = -1 * powerloss_linear_threshold * POWERLOSS_LINEAR_RATE + (powerloss_linear_threshold / POWERLOSS_CUBIC_DIVISOR) ** 3
-
 /obj/machinery/power/supermatter_crystal/examine(mob/user)
 	. = ..()
 	var/immune = HAS_TRAIT(user, TRAIT_MADNESS_IMMUNE) || (user.mind && HAS_TRAIT(user.mind, TRAIT_MADNESS_IMMUNE))
-	if(isliving(user) && !immune && (get_dist(user, src) < HALLUCINATION_RANGE(power)))
+	if(isliving(user) && !immune && (get_dist(user, src) < SM_HALLUCINATION_RANGE(internal_energy)))
 		. += span_danger("You get headaches just from looking at it.")
 	. += delamination_strategy.examine(src)
 	return .
 
+/obj/machinery/power/supermatter_crystal/process_atmos()
+	// PART 1: PRELIMINARIES
+	if(disable_process)
+		return
+
+	var/turf/local_turf = loc
+	if(!istype(local_turf))//We are in a crate or somewhere that isn't turf, if we return to turf resume processing but for now.
+		return  //Yeah just stop.
+	if(isclosedturf(local_turf))
+		var/turf/did_it_melt = local_turf.Melt()
+		if(!isclosedturf(did_it_melt)) //In case some joker finds way to place these on indestructible walls
+			visible_message(span_warning("[src] melts through [local_turf]!"))
+		return
+
+	// PART 2: GAS PROCESSING
+	var/datum/gas_mixture/env = local_turf.return_air()
+	absorbed_gasmix = env?.remove_ratio(absorption_ratio) || new()
+	absorbed_gasmix.volume = (env?.volume || CELL_VOLUME) * absorption_ratio // To match the pressure.
+	calculate_gases()
+	// Extra effects should always fire after the compositions are all finished
+	// Some extra effects like [/datum/sm_gas/carbon_dioxide/extra_effects]
+	// needs more than one gas and rely on a fully parsed gas_percentage.
+	for (var/gas_path in absorbed_gasmix.gases)
+		var/datum/sm_gas/sm_gas = GLOB.sm_gas_behavior[gas_path]
+		sm_gas.extra_effects(src)
+
+	// PART 3: POWER PROCESSING
+	internal_energy_factors = calculate_internal_energy()
+	zap_factors = calculate_zap_multiplier()
+	if(internal_energy && (last_power_zap + 4 SECONDS - (internal_energy * 0.001)) < world.time)
+		playsound(src, 'sound/weapons/emitter2.ogg', 70, TRUE)
+		hue_angle_shift = clamp(903 * log(10, (internal_energy + 8000)) - 3590, -50, 240)
+		var/zap_color = color_matrix_rotate_hue(hue_angle_shift)
+		supermatter_zap(
+			zapstart = src,
+			range = 3,
+			zap_str = 5 * internal_energy * zap_multiplier,
+			zap_flags = ZAP_SUPERMATTER_FLAGS,
+			zap_cutoff = 300,
+			power_level = internal_energy,
+			color = zap_color,
+		)
+		last_power_zap = world.time
+
+	// PART 4: DAMAGE PROCESSING
+	temp_limit_factors = calculate_temp_limit()
+	damage_factors = calculate_damage()
+	if(damage == 0) // Clear any in game forced delams if on full health.
+		set_delam(SM_DELAM_PRIO_IN_GAME, SM_DELAM_STRATEGY_PURGE)
```

Comment:
why is this second arg here if it defaults to it? why not set it on the second use of this proc too?

Reply:
Ill write the full args then for clarity.